pkg/apis/serving/v1alpha1/kfservice_validation.go

```diff
@@ -24,15 +24,13 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+// Known error messages
 const (
-	// MinReplicasShouldBeLessThanMaxError is an known error message
 	MinReplicasShouldBeLessThanMaxError = "MinReplicas cannot be greater than MaxReplicas"
-	// MinReplicasLowerBoundExceededError is an known error message
-	MinReplicasLowerBoundExceededError = "MinReplicas cannot be less than 0"
-	// MaxReplicasLowerBoundExceededError is an known error message
-	MaxReplicasLowerBoundExceededError = "MaxReplicas cannot be less than 0"
-	// TrafficBoundsExceededError is an known error message
-	TrafficBoundsExceededError = "TrafficPercent must be between [0, 100]"
+	MinReplicasLowerBoundExceededError  = "MinReplicas cannot be less than 0"
+	MaxReplicasLowerBoundExceededError  = "MaxReplicas cannot be less than 0"
+	TrafficBoundsExceededError          = "TrafficPercent must be between [0, 100]"
+	TrafficProvidedWithoutCanaryError   = "Canary may not be unspecified if CanaryTrafficPercent is provided."
```

Comment:
How about **Canary must be specified when CanaryTrafficPercent > 0** ?

Reply:
Sounds good